src/sync/sys/windows/net.rs

```diff
@@ -0,0 +1,303 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::error::Result;
+
+use mio::windows::NamedPipe;
+
+use std::ffi::OsStr;
+use std::fs::OpenOptions;
+use std::os::windows::ffi::OsStrExt;
+use std::os::windows::fs::OpenOptionsExt;
+use std::os::windows::io::{FromRawHandle, IntoRawHandle, RawHandle};
+use std::os::windows::prelude::AsRawHandle;
+use std::sync::atomic::{AtomicBool, AtomicI32, Ordering};
+use std::sync::{Arc, Mutex};
+use std::{io};
+
+
+use windows_sys::Win32::Foundation::{ERROR_NO_DATA, INVALID_HANDLE_VALUE, CloseHandle, GetLastError};
+use windows_sys::Win32::Storage::FileSystem::{
+    FILE_FLAG_FIRST_PIPE_INSTANCE, FILE_FLAG_OVERLAPPED, PIPE_ACCESS_DUPLEX,
+};
+use windows_sys::Win32::System::Pipes::{
+    CreateNamedPipeW, PIPE_TYPE_BYTE, PIPE_UNLIMITED_INSTANCES,
+};
+use mio::{Events, Interest, Poll, Token};
+use std::io::{Read, Write};
+
+const SERVER: Token = Token(0);
+const CLIENT: Token = Token(1);
+
+pub struct PipeListener {
+    first_instance: AtomicBool,
+    address: String,
+    instance_number: AtomicI32,
+}
+
+impl PipeListener {
+    pub(crate) fn new(sockaddr: &str) -> Result<PipeListener> {
+        Ok(PipeListener {
+            first_instance: AtomicBool::new(true),
+            address: sockaddr.to_string(),
+            instance_number: AtomicI32::new(1),
+        })
+    }
+
+    pub(crate) fn accept(
+        &self,
+        quit_flag: &Arc<AtomicBool>,
+    ) -> std::result::Result<Option<PipeConnection>, io::Error> {
+        if quit_flag.load(Ordering::SeqCst) {
+            info!("listener shutdown for quit flag");
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                "listener shutdown for quit flag",
+            ));
+        }
+
+        // Create a new pipe for every new client
+        let mut namedpipe = self.new_instance().unwrap();
+
+        let mut poll = Poll::new().unwrap();
+        let mut events = Events::with_capacity(1024);
+        poll.registry()
+            .register(
+                &mut namedpipe,
+                SERVER,
+                Interest::WRITABLE,
+            )
+            .unwrap();
+
+        loop {
+            match namedpipe.connect() {
+                Ok(()) => {
+                    // pipe is locked so can't use it here.
+                    break;
+                }
+                Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                    println!("waiting for client to connect");
+                    poll.poll(&mut events, None).unwrap();
+                    continue;
+                }
+                Err(e) => {
+                    println!("Error connecting to pipe: {e}");
```

Comment:
```suggestion
                    info!("Error connecting to pipe: {e}");
```

Reply:
Although, do we actually need the log now that I'm thinking. At least in this match branch specifically, as the caller can just do something with the error. Can we just add more context to the error instead?

Reply:
I;ve removed the line for now, I am not sure more context is really needed.